src/js/view/controls/settings-menu.js

```diff
@@ -182,4 +186,21 @@ export function setupSubmenuListeners(settingsMenu, controlbar, model, api) {
             onPlaybackRatesChange(model, model.get('playbackRates'));
         }
     });
+
+    model.on('change:streamType', (changedModel, newStreamType, previousStreamType) => {
+        if (newStreamType === previousStreamType) {
+            return;
+        }
+        if (newStreamType === 'LIVE') {
```

Comment:
This is essentially doing the same thing as `onPlaybackRatesChange`. Instead of duplicating code here, I suggest modifying `onPlaybackRatesChange` and ensure that you don't attempt to add the menu if it already exists.